### ISSUE TITLE
Full Path to .py script fails

### ISSUE THREAD

**Issue by isa56k** [extrabacon/python-shell#38]
I'm using the following to call my python script:

```
var PythonShell = require('python-shell');
var pyshell = new PythonShell.run('/home/pi/Development/Sensor/python/SenseHat.py');

// get message back
pyshell.on('message', function (message) {
  // received a message sent from the Python script (a simple "print" statement) 
  console.log(message);
});

// end the input stream and allow the process to exit 
pyshell.end(function (err) {
  if (err) throw err;
  console.log('exit');
});
```

If I use the full path to the script it fails with the error below:

```
/home/pi/Development/Sensor/node/pythonTest.js:14
  if (err) throw err;
           ^
Error: python: can't open file 'home/pi/Development/Sensor/python/SenseHat.py': [Errno 2] No such file or directory

    at PythonShell.parseError (/home/pi/node_modules/python-shell/index.js:190:17)
    at terminateIfNeeded (/home/pi/node_modules/python-shell/index.js:98:28)
    at Socket.<anonymous> (/home/pi/node_modules/python-shell/index.js:78:9)
    at emitNone (events.js:72:20)
    at Socket.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:903:12)
    at doNTCallback2 (node.js:439:9)
    at process._tickCallback (node.js:353:17)

```

If I use the following path to the script it all works ok:
`../python/SenseHat.py
`
and so does:
`
../../../../../home/pi/Development/Sensor/python/SenseHat.py`

**Comment by extrabacon** [comment]
It'a a bug, absolute paths are not supported because of the handling of the `scriptPath` option (https://github.com/extrabacon/python-shell/blob/master/index.js#L53).

Quick workaround: set `scriptPath` option to directory containing your script (/home/pi/Development/Sensor/python), then run the script (SenseHat.py). This way the full path to the script will be built correctly.

**Comment by isa56k** [comment]
```
var pyshell = new PythonShell.run('SenseHat.py', { scriptPath: '/home/pi/Development/Sensor/python/' } );
```

That works, sorry... didn't see that option!

**Comment by extrabacon** [comment]
It's still a workaround, python-shell _should_ support absolute paths. Reopening so I can fix this in a future release.

**Comment by Lakzix** [comment]
Just to know, is it fixed yet? Thanks

**Comment by maziarz** [comment]
Heres a dirty workaround concept for OSX. I recommmend using `os` lib to determine the OS (darwin, win32, etc.)

`
import path from 'path'

const somePath = '/some/path/here/script.py'
let fileName = path.dirname(somePath)
let filePath = path.basename(somePath)

PythonShell.run(fileName, { scriptPath: filePath }, function (err, results) {
    ...
})
`

**Comment by jeremy-harris** [comment]
Just to know, is it fixed yet? Thanks.

(says it is closed; issue is not fixed for me)

**Comment by Almenon** [comment]
? The issue is still open - for now just use the workaround

**Comment by Trevirirus** [comment]
the workaround is not always working:

pyshell = new pyShell('main.py',{scriptPath: path.join(__dirname, '..', 'src', 'GNSS_Positioning')} , options)

also the scriptpath inside the options is resulting in the above described errors.

please support full paths

**Comment by KarthikSamaganam4** [comment]
PythonShell.run('./script.py', options, function(err, data) {
            ^

TypeError: PythonShell.run is not a function
    at Object.<anonymous> (/Users/ksamaganam/IPFS/document_upload/testScript.js:10:13)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3


Getting this error please help me . I am using python2.7 and have installed latest python-shell module

**Comment by Almenon** [comment]
@KarthikSamaganam4 this is unrelated to this issue - I've opened a new issue for you.  See #151

**Comment by Almenon** [comment]
PythonShell should support both relative and absolute paths now.  For example, the runString function works by passing an absolute path to the a file in the temp directory over to pythonshell. The function is unit tested in both windows and mac so it will work no matter the platform.

@Trevirirus, can you double check if it has been fixed in the latest version (1.0.4) of python-shell please? You should be able to pass the path into pythonshell without having to specify any options.

@extrabacon, I fixed it in https://github.com/extrabacon/python-shell/commit/4ba8b5eb93f88fdb3b3b652b5756ec14657f9af0 by removing the ./ which was causing an error with absolute paths on mac.  Was there any reason ./ was in there in the first place? Hopefully I didn't break some weird edge case not covered in the unit tests.
